serverless/serverless
# Add Kinesis and DynamoDB event sources

## What did you implement:

Closes #2365



Added new `-kinesis` and `-dynamodb` events while keeping `-stream` for backward compatibility. This makes it easy for us to detect the type without parsing the ARN. Which means we can take the ARN as it is and pass it to CF which fixes issue #2365

Other than that, the logical ID of the stream is now using an iterator for each type instead of the stream name to drop the dependency on the ARN parsing.

Here's the Pr for the `stream` event: https://github.com/serverless/serverless/pull/2250
## How did you implement it:


## How can we verify it:



To verify, run `serverless deploy --noDeploy` on the following configs and check the update template.
### Verifying old stream syntax works

create a dynamodb table & stream and use the following config:

``` yml
service: serverless-stream-type

provider:
  name: aws
  runtime: nodejs4.3

functions:
  hello:
    handler: handler.hello
    events:
      - stream: arn:aws:dynamodb:us-east-1:123456789:table/stream-type/stream/2016-10-20T15:04:38.834
```
### Verifying dynamodb stream syntax works

create a dynamodb table & stream and use the following config:

``` yml
service: serverless-stream-type

provider:
  name: aws
  runtime: nodejs4.3

functions:
  hello:
    handler: handler.hello
    events:
      - dynamodb: arn:aws:dynamodb:us-east-1:123456789:table/stream-type/stream/2016-10-20T15:04:38.834
```
### Verifying kinesis stream syntax works

create a kinesis stream and use the following config:

``` yml
service: serverless-stream-type

provider:
  name: aws
  runtime: nodejs4.3

functions:
  hello:
    handler: handler.hello
    events:
      - kinesis: arn:aws:kinesis:us-east-1:123456789:table/stream-type/stream/2016-10-20T15:04:38.834
```
## Todos:

**_Is this ready for review?:**_ YES
